src/plot_api/plot_api.js

```diff
@@ -2508,6 +2526,314 @@ Plotly.relayout = function relayout(gd, astr, val) {
 };
 
 /**
+ * Transition to a set of new data and layout properties
+ *
+ * @param {string id or DOM element} gd
+ *      the id or DOM element of the graph container div
+ */
+Plotly.transition = function(gd, data, layout, traceIndices, transitionConfig) {
+    gd = getGraphDiv(gd);
+
+    var i, traceIdx;
+    var fullLayout = gd._fullLayout;
+
+    transitionConfig = Lib.extendFlat({
+        ease: 'cubic-in-out',
+        duration: 500,
+        delay: 0,
+        cascade: 0
+    }, transitionConfig || {});
+
+    // Create a single transition to be passed around:
+    if(transitionConfig.duration > 0) {
+        gd._currentTransition = d3.transition()
```

Comment:
This needs to go. I passed this responsibility down to the trace level so that transitions aren't *all* synchronized. Though that would definitely be a possibility.